src/expr/src/visit.rs

```diff
@@ -56,12 +56,14 @@ pub trait VisitChildren<T> {
     /// Apply a fallible immutable function `f` to each direct child.
     fn try_visit_children<F, E>(&self, f: F) -> Result<(), E>
     where
-        F: FnMut(&T) -> Result<(), E>;
+        F: FnMut(&T) -> Result<(), E>,
+        E: From<RecursionLimitError>;
```

Comment:
I was confused about why the `RecursionLimitError` is needed here, since we only visit direct children without recursing. The answer is that we might need to recurse to find children that are inside fields of the mutually recursive type. It would be nice to have a mention of that in the docstring.